pylabrobot/pumps/calibration.py

```diff
@@ -0,0 +1,181 @@
+from __future__ import annotations
+
+import json
+import csv
+from typing import Union, Dict, List, Optional
+
+
+class PumpCalibration:
+  """ Calibration for a single pump or pump array
+
+  Attributes:
+    calibration: The calibration of the pump or pump array.
+  """
+
+  def __init__(self, calibration: Optional[List[Union[float, int]]] = None):
+    """ Initialize a PumpCalibration object.
+
+    Args:
+      calibration: calibration of the pump in pump-specific volume per time/revolution units.
+
+    Raises:
+      ValueError: if a value in the calibration is outside expected parameters.
+    """
+
+    if calibration is not None and any(value <= 0 for value in calibration):
+      raise ValueError("A value in the calibration is is outside expected parameters.")
+    self.calibration = calibration
+
+  def __getitem__(self, item) -> Union[float, int]:
+    if self.calibration is None:
+      raise TypeError(
+        "Pump is not calibrated. Volume based pumping and related functions unavailable.")
+    return self.calibration[item]  # type: ignore
+
+  def __bool__(self):
+    return self.calibration is not None
```

Comment:
I don't think this is still needed, right?

Also, I think the calibration attribute should never be None on this class - it is cleaner to have the PumpCalibration variable/attribute elsewhere be None than to initialize it to a 'useless' (in that context) object.